lib/init.lua

```diff
@@ -123,46 +126,48 @@ local function hash(arr): string | number
 end
 
 local function createArchetypeRecords(componentIndex: ComponentIndex, to: Archetype, from: Archetype?)
-	local destinationCount = #to.types
 	local destinationIds = to.types
+	local records = to.records
+	local id = to.id
 
-	for i = 1, destinationCount do
-		local destinationId = destinationIds[i]
+	for i, destinationId in destinationIds do
+		local archetypesMap = componentIndex[destinationId]
 
-		if not componentIndex[destinationId] then
-			componentIndex[destinationId] = { size = 0, sparse = {} }
+		if not archetypesMap then
+			archetypesMap = {size = 0, sparse = {}}
+			componentIndex[destinationId] = archetypesMap
 		end
 
-		local archetypesMap = componentIndex[destinationId]
-		archetypesMap.sparse[to.id] = i
-		to.records[destinationId] = i
+		archetypesMap.sparse[id] = i
+		records[destinationId] = i
 	end
 end
 
-local function archetypeOf(world: World, types: { i24 }, prev: Archetype?): Archetype
+local function archetypeOf(world: World, types: {i24}, prev: Archetype?): Archetype
 	local ty = hash(types)
 
-	world.nextArchetypeId = (world.nextArchetypeId::number)+ 1
-    local id = world.nextArchetypeId
+	local id = world.nextArchetypeId + 1
+	world.nextArchetypeId = id
 
-	local columns = {} :: { any }
+	local length = #types
+	local columns = table.create(length) :: {any}
 
-	for _ in types do
-		table.insert(columns, {})
+	for index in types do
```

Comment:
I think table.insert is the common idiom here. Unless there is good reason not to, I think it should use that.

Reply:
Was part of the optimizations B^)